src/utilities/routeRegex.ts

```diff
@@ -14,9 +14,23 @@ export function generateRouteQueryRegexPatterns(route: RouterRoute): RegExp[] {
     .map(([key, value]) => new RegExp(`${key}=${replaceParamSyntaxWithCatchAlls(value)}`, 'i'))
 }
 
-function replaceParamSyntaxWithCatchAlls(value: string): string {
+export function replaceParamSyntaxWithCatchAlls(value: string): string {
+  return [
+    replaceOptionalParamSyntaxWithCatchAll,
+    replaceRequiredParamSyntaxWithCatchAll,
+  ].reduce((pattern, regexBuild) => {
+    return regexBuild(pattern)
+  }, value)
+}
```

Comment:
this file really only needed to update capture groups to be non-capturing. I decided to go the extra mile of moving "regex building" string replace methods into private functions with a reduce to reuse the pattern I'm now using in paramsFinder.ts